K/KaHyPar/build_tarballs.jl

```diff
@@ -3,46 +3,46 @@
 using BinaryBuilder, Pkg
 
 name = "KaHyPar"
-version = v"1.0.0"
+version = v"1.2.1"
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://github.com/arbenson/kahypar.git", "d64498068ab745a09a94391e746d35853fc3eef3"),
-    DirectorySource("./bundled"),
+    GitSource("https://github.com/kahypar/kahypar.git", "971c38a4c7b7ed1ecf001d0263c1099cbc72886e"),
+    DirectorySource("./bundled")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/kahypar/
-git submodule update --init --recursive
-atomic_patch -p1 ../patches/disable-tests.patch
+cd $WORKSPACE/srcdir
+for f in ${WORKSPACE}/srcdir/patches/*.patch; do
+    atomic_patch -p1 ${f}
+done
+cd kahypar/
+git submodule update --init --recursive --depth=1
 mkdir build && cd build
 cmake .. -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
-make -j${nproc}
-make install
+make install.library
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = [
-    Platform("i686", "linux"; libc="glibc"),
-    Platform("x86_64", "linux"; libc="glibc"),
-    Platform("i686", "linux"; libc="musl"),
-    Platform("x86_64", "linux"; libc="musl"),
-    Platform("x86_64", "macos"),
-    Platform("x86_64", "freebsd"),
-    Platform("i686", "windows"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("x86_64", "macos"; ),
+    Platform("x86_64", "freebsd"; )
 ]
-platforms = expand_cxxstring_abis(platforms)
+
 
 # The products that we will ensure are always built
 products = [
     LibraryProduct("libkahypar", :libkahypar)
 ]
 
 # Dependencies that must be installed before this package can be built
-dependencies = Dependency[
+dependencies = [
+    Dependency(PackageSpec(name="boost_jll", uuid="28df3c45-c428-5900-9ff8-a3135698ca75"))
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"6.1.0")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"10.2.0")
```

Comment:
I don't recommend at all using GCC 10.  In general use the oldest version of GCC possible, for maximum compatibility.  If GCC 6 is enough to build the library, we should stick to it

Reply:
OK, I think kahypar recommends GCC 9, should we use GCC 9 then?

Reply:
Recommends or requires?  Everybody recommends the latest compiler, but we want the largest compatibility possible, which is given by the oldest GCC possible

Reply:
it says "requirements" is GCC 9 or higher so I guess it is required: https://github.com/kahypar/kahypar#requirements

Reply:
but it seems GCC supports C++17 since GCC 5: https://gcc.gnu.org/projects/cxx-status.html#cxx17

Reply:
also should we remove the julia compat, I ran the wizard on julia 1.6, but I guess this is not required.

Reply:
OK GCC 7 seems to be the lowest GCC version that works